### ISSUE TITLE
MTA compile warnings: overriding /MTd with /MT

### ISSUE THREAD

**Issue by qaisjp** [multitheftauto/mtasa-blue#1736]
**Describe the bug**

In debug mode we provide both the `/MT` and `/MTd` flags. We should only be providing `/MTd` when compiling Debug. **Edit: or, due to https://github.com/multitheftauto/mtasa-blue/issues/1736#issuecomment-711329132, we should never be providing `/MTd`.**

See [/MD, /MT, /LD (Use Run-Time Library) @ docs.microsoft.com](https://docs.microsoft.com/en-us/cpp/build/reference/md-mt-ld-use-run-time-library?view=vs-2019) for more information on what these flags mean.

I think this is a good small task for anyone interested in learning how Premake works.

**To reproduce**

Build MTA

**Expected behaviour**

No warnings

**Screenshots**

![image](https://user-images.githubusercontent.com/923242/96373910-55940600-1167-11eb-9745-85bb7e24c37a.png)

**Comment by ccw808** [comment]
I think there was a compatibility problem with the net module when using debug runtime

**Comment by sbx320** [comment]
Yup. The issue was regarding memory layout being different in some C++ standard containers (primarily `std::string`) between Debug & Release builds. As we only provide release builds of net, we need to use the release standard containers everywhere.

When I looked at this issue in the past, it was caused by /MTd being added by premake when setting up a debug build, Fixing that involved adding support for our special case to premake itself.
